### PR TITLE
Extracting MessageID and Process() to base class

### PR DESCRIPTION
**Pull request by TheLogster**
General code cleanup:

Adding enum for message type ids
Renaming field attribute to something more obvivous
Base class implementation of IMessage to reduce code repeats of MessageID property and empty Process() implementation

### CODE REVIEW

File: src/PFire.Core/Protocol/XMessageField.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Text;
+
+namespace PFire.Protocol
+{
+    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Method | AttributeTargets.Property, AllowMultiple = true)]
+    internal sealed class XMessageField : Attribute
+    {
+        public string Name { get; }
+        public byte[] NameAsBytes { get; }
```
Comment (darcymiranda) [inline src/PFire.Core/Protocol/XMessageField.cs]: A small concern with this is someone can accidentally diverage the value between `Name` and `NameAsBytes`. I'd prefer `NameAsBytes => Encoding.UTF8.GetBytes(name)`

File: src/PFire.Core/Protocol/Messages/XFireMessageType.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+﻿
+namespace PFire.Core.Protocol.Messages
+{
+    public enum XFireMessageType : short
```
Comment (darcymiranda) [inline src/PFire.Core/Protocol/Messages/XFireMessageType.cs:4]: 🙏

**Review by darcymiranda — APPROVED** [review]
Thank you for taking the time to clean up the code base a little. This was all written many years ago and needs a lot of these types of approvements, so this is a great start!

I'll accept the `Context` -> `XFireClient` change. I went with `Context` for similarity with the `HttpContext` pattern used in the dotnet core http framework.

File: src/PFire.Core/Protocol/XMessageField.cs (hunk 1 above)
Comment (TheLogster) [reply src/PFire.Core/Protocol/XMessageField.cs]: Yep, good call. I have changed NameAsBytes to your suggestion.

File: src/PFire.Core/Protocol/Messages/XFireMessageType.cs (hunk 2 above)
Comment (TheLogster) [reply src/PFire.Core/Protocol/Messages/XFireMessageType.cs:4]: ;)

**Pull request merged**